test/plots/waterfallPlotTests.ts

```diff
@@ -1,123 +1,251 @@
-        let previousValue = yAccessor(data[previousIndex], previousIndex, dataset);
-        if (previousTotal) {
-          if (currentTotal || currentValue < 0) {
-            assert.isTrue(+previousBar.attr("y") === +currentBar.attr("y"), "bars are top/top aligned");
-          } else {
-            assert.closeTo(+previousBar.attr("y") - (+currentBar.attr("y") + +currentBar.attr("height")), 0, 1,
-              "bars are top/bottom aligned");
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        dataset = new Plottable.Dataset(growthBarData);
+        xScale = new Plottable.Scales.Category();
+        yScale = new Plottable.Scales.Linear();
+        plot = new Plottable.Plots.Waterfall<string, number>();
+        plot.x((d) => d.x, xScale);
+        plot.y((d) => d.y, yScale);
+        plot.total((d, i) => i === 0);
+        plot.addDataset(dataset);
+        plot.renderTo(svg);
+      });
+
+      it("classes growth bars", () => {
+        let bars = plot.content().selectAll("rect");
+        assert.strictEqual(bars.size(), growthBarData.length, "all bars are growth except for first");
+        bars.each(function(d, i) {
+          if (i === 0) {
+            return;
           }
-        } else {
-          if (previousValue > 0) {
-            if (currentTotal || currentValue < 0) {
-              assert.isTrue(+previousBar.attr("y") === +currentBar.attr("y"), "bars are top/top aligned");
-            } else {
-              assert.closeTo(+previousBar.attr("y") - (+currentBar.attr("y") + +currentBar.attr("height")), 0, 1,
-                "bars are top/bottom aligned");
-            }
-          } else {
-            if (currentTotal || currentValue < 0) {
-              assert.closeTo(+previousBar.attr("y") + +previousBar.attr("height") - +currentBar.attr("y"), 0, 1,
-                "bars are bottom/top aligned");
-            } else {
-              assert.closeTo((+previousBar.attr("y") + +previousBar.attr("height")) -
-                (+currentBar.attr("y") + +currentBar.attr("height")), 0, 1, "bars are bottom/bottom aligned");
-            }
+          let bar = d3.select(this);
+          assert.isTrue(bar.classed("waterfall-growth"), "bar classed as growth bar");
+        });
+        plot.destroy();
+        svg.remove();
+      });
+
+      it("bars placed at current sum", () => {
+        let bars = plot.content().selectAll("rect.waterfall-growth");
+        assert.strictEqual(bars.size(), growthBarData.length - 1, "all bars are growth except for first");
+        let yAccessor = plot.y().accessor;
+        let sum = 0;
+        bars.each(function(d, i) {
+          let dataY = yAccessor(d, i, dataset);
+          let bar = d3.select(this);
+          assert.closeTo(numAttr(bar, "y") + numAttr(bar, "height"), yScale.scale(sum),
+            window.Pixel_CloseTo_Requirement, "growth bar top at final sum");
+          sum += dataY;
```

Comment:
How do you mean?  I am trying to isolate growth from decline from total.  How should I include total/delta?